### ISSUE TITLE
Unable to decipher error

### ISSUE THREAD

**Issue by dataskills** [neo4j-php/neo4j-php-client#57]
I am trying out this library with a neo4j aura instance. I can't get it working and I have not been able to understand the error I am getting, any insight would be much appreciated, this is what I am getting:

```
Exception 'Laudis\Neo4j\Exception\Neo4jException' with message 'Neo4j errors detected. First one with code "" and message "Read error"'

in /Users/dataskills/Sites/localhost/myApp/vendor/laudis/neo4j-php-client/src/Bolt/Session.php:125

Stack trace:
#0 /Users/dataskills/Sites/localhost/myApp/vendor/laudis/neo4j-php-client/src/Bolt/Session.php(78): Laudis\Neo4j\Bolt\Session->beginTransaction(NULL, NULL)
#1 /Users/dataskills/Sites/localhost/myApp/vendor/laudis/neo4j-php-client/src/Bolt/Session.php(73): Laudis\Neo4j\Bolt\Session->openTransaction()
#2 /Users/dataskills/Sites/localhost/myApp/vendor/laudis/neo4j-php-client/src/Bolt/Session.php(83): Laudis\Neo4j\Bolt\Session->runStatements(Array)
#3 /Users/dataskills/Sites/localhost/myApp/vendor/laudis/neo4j-php-client/src/Bolt/Session.php(88): Laudis\Neo4j\Bolt\Session->runStatement(Object(Laudis\Neo4j\Databags\Statement))
#4 /Users/dataskills/Sites/localhost/myApp/vendor/laudis/neo4j-php-client/src/Client.php(74): Laudis\Neo4j\Bolt\Session->run('CREATE (n:Perso...', Array)
#5 /Users/dataskills/Sites/localhost/myApp/commands/Neo4jController.php(83): Laudis\Neo4j\Client->run('CREATE (n:Perso...')
...
```
I am just trying to run a simple `CREATE` statement to test things out and I am getting the exception with no code and a "Read Error". The error does not contain any useful description. The node was not created (I checked using the browser), so I don't even know what it's tryig to read.

**Comment by dataskills** [comment]
After further review, I can see that the function `unpackProtocolVersion()` is failing at line 161 when it is trying to read the protocol version from the connection (I think).

**Comment by transistive** [comment]
Hello @dataskills,

Thank you for posting this issue. It looks like a more complex issue. I will first have to acquire access to an aura instance. Forgive me, but this issue might take me a few days to fix.

In the meantime, can you check if the HTTP protocol works at least?

Kind regards,

Ghlen

**Comment by fbiville** [comment]
@dataskills Are you using Aura Free, Aura Professional or Aura Enterprise?

**Comment by fbiville** [comment]
Also, what connection URI (scheme) are you using, @dataskills?

**Comment by dataskills** [comment]
I am using Aura Free to test the service. Forgive me if I am mistaken but it looks like the only scheme I can use is the neo4j scheme. See connection URI in the image:
<img width="592" alt="Screen Shot 2021-07-01 at 8 26 37 AM" src="https://user-images.githubusercontent.com/4392698/124150496-79258500-da46-11eb-927a-6b11a2766b33.png">

Is this type of connection not supported? My goal is to find a hosting service, if neo4j aura is not viable can you suggest an alternative? I appreciate your responses @fbiville and @transistive .

**Comment by fbiville** [comment]
It's not completely supported as of now, but I know @transistive is actively working on it.
I'll let him give a more definitive answer.

**Comment by dataskills** [comment]
Ok, thank you. I hope @transistive can shed some more light on this when he has time. In the meantime can you suggest an alternative to neo4j aura for hosting?

**Comment by transistive** [comment]
Hi @dataskills. We have been working the entire day to find the issue and am happy to report we have found the culprit! It was due to some complex SSL configuration which only aura needed to function correctly.

You can expect the bugfix within a few hours. :+1: 

Thanks again for bringing this to our attention.

**Comment by dataskills** [comment]
Very cool, I appreciate it. I look forward to the fix. Cheers.

**Comment by transistive** [comment]
The bug is fixed! Feel free to update to version 2.0.4 to enjoy neo4j aura and php

**Comment by dataskills** [comment]
The fix works! Thank you @transistive, great work!
